packages/protocol/contracts/stability/Reserve.sol

```diff
@@ -509,6 +634,48 @@ contract Reserve is
     return getUnfrozenBalance().add(getOtherReserveAddressesGoldBalance());
   }
 
+  /**
+   * @notice Returns the amount of particular asset included in the reserve.
+   * @return The amount of particular asset included in the reserve.
+   */
+  function getCollateralAssetBalance(address collateralAssetAddress) public view returns (uint256) {
+    return IERC20(collateralAssetAddress).balanceOf(address(this));
+  }
+
+  /**
+   * @notice Returns the amount of particular collateral asset included in other reserve addresses.
+   * @return The particular collateral asset amount included in other reserve addresses.
+   */
+  function getReserveAddressesCollateralAssetBalance(address collateralAssetAddress)
+    public
+    view
+    returns (uint256)
+  {
+    uint256 reserveCollateralAssetBalance = 0;
+    for (uint256 i = 0; i < otherReserveAddresses.length; i++) {
+      reserveCollateralAssetBalance = reserveCollateralAssetBalance.add(
+        IERC20(collateralAssetAddress).balanceOf(otherReserveAddresses[i])
+      );
+    }
+    return reserveCollateralAssetBalance;
+  }
+
+  function addCollateralAsset(address collateralAssetAddress) onlyOwner {
+    require(
+      getCollateralAssetBalance(collateralAssetAddress) > 0,
+      "Token Address specified is not a reserve collateral asset"
+    );
+    isCollateralAsset[collateralAssetAddress] = true;
```

Comment:
Should you also be pushing the address to the array of collateral assets defined on line 62?